lib/query.js

```diff
@@ -1632,6 +1632,9 @@ Query.prototype.setOptions = function(options, overwrite) {
     this._mongooseOptions.defaults = options.defaults;
     // deleting options.defaults will cause 7287 to fail
   }
+  if (this.schema && 'lean' in this.schema.options) {
```

Comment:
Do we also need to check if the user is passing a lean value?
```suggestion
  if (options.lean == null && this.schema && 'lean' in this.schema.options) {
```